### PR TITLE
Add a github action for user deletion

### PR DESCRIPTION
**Pull request by lindseydew**
## What does this change?
Adds a script to build the projects on Github Actions so we can migrate away from Teamcity. 

Tested deploying on to CODE and the deploy works ok 

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: .github/workflows/ci-mobile-save-for-later-user-deletion.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: CI
+on: push
```
Comment (mbd0910) [inline .github/workflows/ci-mobile-save-for-later-user-deletion.yaml:2]: I notice that for all the Personalisation repos, we just have a push trigger here, whereas several of the examples from the migration guide have explicit branch and pull request sections. Do you know what the difference is and is that something we should change?

File: .github/workflows/ci-mobile-save-for-later.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        uses: actions/setup-java@v3
+        with:
+          java-version: 8
+          distribution: temurin
+          cache: sbt
+
+      - name: Create artifacts
+        run: sbt "project mobile-save-for-later" assembly
+
+      - name: Build CDK
```
Comment (mbd0910) [inline .github/workflows/ci-mobile-save-for-later.yaml:26]: I notice that the equivalent `ci-mobile-save-for-later-user-deletion.yaml` file doesn't have this section. Just for my own understanding, why is this needed for mobile S4L but not for the user deletion?

**Review by mbd0910 — COMMENTED** [review]
I'm not a TeamCity expert so apologies for coming with questions rather than suggestions, but having reviewed the migration guide and compared the yaml files, thanks to the testing you've done I'm happy that this looks good.

You also mentioned you had worked out how to test against a running mobile S4L service. Is it worth documenting this somewhere or do you have a plan for that?

**Comment by lindseydew** [comment]
> I'm not a TeamCity expert so apologies for coming with questions rather than suggestions, but having reviewed the migration guide and compared the yaml files, thanks to the testing you've done I'm happy that this looks good.
> 
> You also mentioned you had worked out how to test against a running mobile S4L service. Is it worth documenting this somewhere or do you have a plan for that?

Yeah, I've added some testing notes in the next commit

File: .github/workflows/ci-mobile-save-for-later-user-deletion.yaml (hunk 1 above)
Comment (Mark-McCracken) [reply .github/workflows/ci-mobile-save-for-later-user-deletion.yaml:2]: Sure, can do. I normally say on pushes to main, or pull requests, which covers the vast majority, but doesn't build if you're just saving a ton of commits on the regular before you're ready to PR them

File: .github/workflows/ci-mobile-save-for-later-user-deletion.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+jobs:
+  mobile-save-for-later-user-deletion-ci:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+      - uses: aws-actions/configure-aws-credentials@v1
```
Comment (Mark-McCracken) [inline .github/workflows/ci-mobile-save-for-later-user-deletion.yaml:12]: Might wanna bump this, shouldn't make any noticable difference, but will remove some warnings from older version of AWS javascript sdk which is deprecated
```suggestion
      - uses: aws-actions/configure-aws-credentials@v4
```

Comment (Mark-McCracken) [reply .github/workflows/ci-mobile-save-for-later-user-deletion.yaml:12]: Likewise for other items below

**Review by Mark-McCracken — APPROVED** [review]
Lovely, just a few notes from me, feel free to ignore

**Pull request merged**